compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BResourceFunction.java

```diff
@@ -38,6 +38,7 @@ public class BResourceFunction extends BAttachedFunction {
     public List<Name> resourcePath;
     public Name accessor;
     public BTupleType resourcePathType;
+    public List<BResourcePathSegmentSymbol> pathSegmentSymbols;
```

Comment:
Instead of keeping three fields; `resourcePath`, `resourcePathType`, and `pathSegmentSymbols`, we can only keep pathSegmentSymbols, because others can be inferred from the pathSegmentSymbols. We can keep others as caching values, but if they are updated in a later phase, better to remove those to avoid additional maintenance overhead.